dvc/cache/__init__.py

```diff
@@ -65,18 +65,23 @@ def __init__(self, repo):
 
         self._cache[Schemes.LOCAL] = _get_cache(repo, settings)
 
-        for scheme in self.CLOUD_SCHEMES:
+    def _initalize_cloud_cache(self, schemes):
+        for scheme in schemes:
             remote = self.config.get(scheme)
             settings = {"name": remote} if remote else None
-            self._cache[scheme] = _get_cache(repo, settings)
+            self._cache[scheme] = _get_cache(self.repo, settings)
 
     def __getattr__(self, name):
+        if name not in self._cache and name in self.CLOUD_SCHEMES:
+            self._initalize_cloud_cache([name])
+
         try:
             return self._cache[name]
         except KeyError as exc:
             raise AttributeError from exc
 
     def by_scheme(self):
+        self._initalize_cloud_cache(self.CLOUD_SCHEMES)
```

Comment:
We can safely skip this. In all use-cases of `by_scheme` we filter-out None's anyway.